src/js/components/AppListComponent.jsx

```diff
@@ -186,7 +198,11 @@ var AppListComponent = React.createClass({
 
     if (filterText != null && filterText !== "") {
       nodesSequence = nodesSequence
-        .filter(app => app.id.indexOf(filterText) !== -1);
+        .filter(app => {
+          app.filterScore = score(app.id, filterText);
```

Comment:
Alphabetically sorting is done before. 
https://github.com/mesosphere/marathon-ui/blob/master/src/js/components/AppListComponent.jsx#L315

The sort-function mentioned above should be added below that.

I did a mistake in the condition, must be:
```
if (filterText != null && sortKey === "id") {
```